Classes/Widgets/DuplicateFilesWidget.php

```diff
@@ -14,23 +13,24 @@
 class DuplicateFilesWidget implements WidgetInterface
 {
     public function __construct(
-        private ?ConnectionPool $connectionPool = null,
-        private ?StandaloneView $view = null,
-        private ?WidgetConfigurationInterface $configuration = null,
+        private ConnectionPool $connectionPool,
+        private ResourceFactory $resourceFactory,
+        private StandaloneView $view,
+        private WidgetConfigurationInterface $configuration,
         private readonly array $options = []
     )
     {}
 
     public function renderWidgetContent(): string
     {
-        $resourceFactory = GeneralUtility::makeInstance(ResourceFactory::class);
         $queryBuilder = $this->connectionPool->getConnectionForTable('sys_file')->createQueryBuilder();
         $queryBuilder->getRestrictions()->removeAll();
 
         $duplicates = $queryBuilder
             ->selectLiteral('GROUP_CONCAT(uid) as uids', 'sha1', 'count(*) as counting')
             ->from('sys_file')
-            ->addOrderBy('counting', 'desc')
+            ->orderBy('counting', 'desc')
+            ->addOrderBy('size', 'desc')
             ->where('missing = 0 AND storage > 0 AND name != "index.html" AND identifier NOT LIKE "%_recycler_%"')
```

Comment:
->where('missing = 0 AND storage > 0 AND name != "index.html" AND identifier NOT LIKE "%_recycler_%"')

`missing = 0` only works if index scheduler is executed.
I got errors where file is not marked as missing but Folder doesn't exists anymore.

Reply:
then, we have to fix the folder errors as missing = 0 is default for sys_file